src/sentry_value.c

```diff
@@ -431,6 +431,34 @@ sentry__value_new_object_with_size(size_t size)
     }
 }
 
+sentry_value_t
+sentry_value_new_user(const int32_t *id, const char *username,
+    const char *email, const char *ip_address)
```

Comment:
I'm not 💯 on passing in a pointer for the `id` here, but if we change the signature to be 
```c
sentry_value_new_user(int32_t id, const char *username, const char *email, const char *ip_address)
```
instead, the `id` can never be NULL, which would mean we enforce it to not be optional (while the [docs](https://docs.sentry.io/platforms/native/enriching-events/identify-user/) say each of them should be).

Reply:
It is your lucky day: `id` is also a string.